### PR TITLE
Javalab: Upgrade codemirror

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Upgrades codemirror and all related sub-packages to their latest versions (0.19.9 and below). There was just one breaking change in how newlines and indents were handled.

## Links

JIRA: https://codedotorg.atlassian.net/browse/CSA-879

## Testing story

Tested locally on all Javalab All The Things Levels as well as projects/javalab/new in both light and dark mode. Behavior appears to be consistent with production.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by jmkulwik — APPROVED** [review]
Generally looks good. Does tabbing still add 2 spaces, or will this now add tabs instead?

**Comment by sanchitmalhotra126** [comment]
> Generally looks good. Does tabbing still add 2 spaces, or will this now add tabs instead?

Yep, it still adds two spaces.

File: apps/package.json [hunk 1]
```diff
@@ -83,7 +83,7 @@
     "chalk": "^1.1.3",
     "chrome-serialport": "bcjordan/chrome-serialport#2",
     "classnames": "^2.2.5",
-    "codemirror": "5.5",
+    "codemirror": "5.63.1",
```
Comment (molly-moen) [inline apps/package.json]: as an fyi, this codemirror version is not used by csa (I think it's only used by some internal tools). If there aren't any breaking changes it's probably fine to upgrade it though!

**Comment by bencodeorg** [comment]
This may be unnecessary, but considering it's still in beta, could you check that on s/allthethings/lessons/44/levels/6 (a read only mode level), you're unable to paste text into the codemirror? We did some work a couple months ago to bump codemirror to get that to work, just want to make sure it does not regress (pretty sure it's not tested).

File: apps/package.json (hunk 1 above)
Comment (jmkulwik) [reply apps/package.json]: It gets used by levelbuilder, which doesn't tend to have test coverage. There may be breaking changes that don't notify us.

Comment (sanchitmalhotra126) [reply apps/package.json]: @jmkulwik Ah okay, could you clarify how this gets used by levelbuilder? What would be the best way to test this to make sure this isn't breaking anything on levelbuilder?

**Comment by sanchitmalhotra126** [comment]
> This may be unnecessary, but considering it's still in beta, could you check that on s/allthethings/lessons/44/levels/6 (a read only mode level), you're unable to paste text into the codemirror? We did some work a couple months ago to bump codemirror to get that to work, just want to make sure it does not regress (pretty sure it's not tested).

Good catch, looks like this does allow pasting now which is a regression (but still not typing). Would you be able to point me to the previous work that was done here? I'll see how best to not regress this behavior with the upgrade.

**Comment by bencodeorg** [comment]
I think these two PRs have everything -- I remember CodeMirror being pretty confusing to understand, let me know if you want to look at it together (not that I remember any of it any more...)

https://github.com/code-dot-org/code-dot-org/pull/41273
https://github.com/code-dot-org/code-dot-org/pull/41301

File: apps/package.json (hunk 1 above)
Comment (jmkulwik) [reply apps/package.json]: I'd probably consider any updates to our usage of codemirror 5 to be a different feature set. They're pretty different editors used in very different parts of our website. I'd recommend doing this in a separate PR if you want to make this change. 

To hand test it, go to /levels/new, create any coding level (App Lab would work well) and enter text into a code field on the level edit form.

Comment (sanchitmalhotra126) [reply apps/package.json]: Sounds good, I'll revert this for now

**Comment by sanchitmalhotra126** [comment]
> I think these two PRs have everything -- I remember CodeMirror being pretty confusing to understand, let me know if you want to look at it together (not that I remember any of it any more...)
> 
> #41273 #41301

Thanks, this was helpful! This latest commit should fix the readonly regression.

**Review by bencodeorg — APPROVED** [review]
Nice!

**Pull request merged**